### DISCUSSION TITLE
Error saving datasource: ECONNECTION Refused trying to ssh tunnel to an AWS RDS MySQL 5.7.33 DB through AWS EC2

### DISCUSSION THREAD

**Discussion by ralflukner** [Budibase/budibase#3167] [Help]
I set up port forwarding to be able to ssh tunnel to my AWS RDS MySQL DB.  I'm able to get SQLPro for MySQL and other apps such as JetBrains applications to connect to the database through localhost:3306->EC2:3306->MySQL:3306.  However, when I try to connect to use budibase to create a MySQL datasource at localhost:3306 I get an Error saving datasource: ECONNECTION Refused error.

**Comment by shogunpurple** [comment]
Hi @ralflukner, I think you may be experiencing this because budibase runs inside of docker, and your docker containers don't have access to the host machines network on `localhost`.

Try the suggestions in this comment: https://github.com/Budibase/budibase/discussions/2954#discussioncomment-1463535 and let me know if they resolve your issue!

**Reply by ralflukner** [reply]
Unfortunately I cannot run the database server locally without crippling the local machine because the database is too CPU and memory intensive (large amount of data that is frequently accessed and changed).
RBL

**Reply by shogunpurple** [reply]
Yeah thats fine! I just mean that `localhost` is possibly not resolving to the SSH tunnel you have set up on your local machine because budibase is running inside docker. If you change the host to `host.docker.internal` or `172.17.0.1` on linux, it should hopefully connect to your local SSH tunnel.
